resolvers/Mutation.js

```diff
@@ -112,8 +112,17 @@ const Mutation = {
         }
         try{
             let sql = `UPDATE \`Item_Trans\`
-            SET \`name\` = '${data.name}'
-            WHERE \`itemId\`= '${currentItem.id}'`
+            SET \`name\` = '${data.name}', \`description\` = '${data.description}'
+            WHERE \`itemId\`= '${currentItem.id}' AND \`lang\`= 'zh'`
+            await dbMutation(sql)
+        }catch(e){
+            console.log("fail update name", e)
+        }
+        // it seems like we don't have english type column in DB
```

Comment:
why? we have type

Reply:
but the problem here if the kitchen only wants to update one field, for example **englishName**, then your query will fail right? Since it requires name, description and type all to be passed

Reply:
I see. And modified that.

Reply:
I remember frontend stuff told me that he pass empty string if the frontend form is not filled completely.

Reply:
then your entire query will crash

Reply:
so then you'd set empty name, since it's empty string

Reply:
Their mutation request will fail. But our server doesn't crash, because we use try catch when querying database.

Reply:
what i mean is they for example pass
description "",
name: "someName",
type: ""

and your query will set description and type to empty string for the database

Reply:
You're correct!

Reply:
not sure what's the proper way to fix it aside from adding many **IF** statements haha

Reply:
so i guess as a temporary solution you can add many **IF** statements like 
```
        if (data.name !== undefined && data.name !== "") {
            try{
                let sql = `UPDATE \`Item_Trans\`
            SET \`name\` = '${data.name}'
            WHERE \`itemId\`= '${currentItem.id}' AND \`lang\`= 'zh'`
                await dbMutation(sql)
            }catch(e){
                console.log("fail update name", e)
            }
        }
        
        if (data.description !== undefined && data.description !== "") {
            try{
                let sql = `UPDATE \`Item_Trans\`
            SET \`description\` = '${data.description}'
            WHERE \`itemId\`= '${currentItem.id}' AND \`lang\`= 'zh'`
                await dbMutation(sql)
            }catch(e){
                console.log("fail update name", e)
            }
        }

等等
```

Reply:
It's more easy to do that with react useState.

Reply:
if they add such logic then it's fine you can merge like this

Reply:
If we want to deal with it in backend. We have to for loop all attribute of data(in schema) to see if it's empty string or not. And query db for each attribute of data(in schema).